src/connectors/keepeek/connector.ts

```diff
@@ -0,0 +1,536 @@
+import { Connector, Media } from '@chili-publish/studio-connectors';
+
+interface MediaLinks {
+  large: string;
+  medium: string;
+  preview: string;
+  small: string;
+  whr: string;
+  xlarge: string;
+}
+interface KeepeekLink {
+  href: string;
+}
+
+interface KeepeekMediaItem {
+  id: number;
+  title: string;
+  statusUpdateDate: string;
+  permission: string;
+  mediaType: string;
+  fileSize: number;
+  fileSizeString: string;
+  updateDate: string;
+  duplicationStatus: string;
+  thumbnailGenerationStatus: string;
+  _links: KeepeekMediaLinks;
+}
+
+interface KeepeekMediaLinks {
+  self: KeepeekLink;
+  'kpk:large'?: KeepeekLink;
+  'kpk:medium'?: KeepeekLink;
+  preview?: KeepeekLink;
+  'kpk:small'?: KeepeekLink;
+  'kpk:whr'?: KeepeekLink;
+  'kpk:xlarge'?: KeepeekLink;
+}
+
+export default class KeepeekConnector implements Media.MediaConnector {
+  private runtime: Connector.ConnectorRuntimeContext;
+
+  // Get the DataCache singleton
+  private DataCache = DataCache.getInstance();
+
+  // Get the KeepeekHelper singleton
+  private KeepeekHelper = new KeepeekHelper();
+
+  private allowedMimeType: string[] = [
+    'application/pdf', // PDF
+    'image/jpeg', // JPEG
+    'image/png', // PNG
+    'image/gif', // GIF
+    'image/svg+xml', // SVG
+    'image/tiff', // TIFF
+  ];
+
+  constructor(runtime: Connector.ConnectorRuntimeContext) {
+    this.runtime = runtime;
+  }
+
+  async query(
+    options: Connector.QueryOptions,
+    context: Connector.Dictionary
+  ): Promise<Media.MediaPage> {
+    // When pageSize is 1 & collection is null, we know that query is called before download
+    if (options.pageSize == 1 && !options.collection) {
```

Comment:
This is incorrect. We don't specify `pageSize as 1` in this situation. You can rely on empty `options.collection` and  `length === 1`
```suggestion
    if (!options.collection && options.filter.length === 1) {
```

Reply:
It's in your documentation : https://docs.chiligrafx.com/GraFx-Developers/connectors/media-connector/build-a-simple-media-connector/

**Fixing query**
When query is called before download, the options.pageSize will be the value 1 and options.collection will be null.

Reply:
Correct. Ok. please keep it as is so far. We will figure out with a team about approach

Reply:
UPDATE: We will fix it from our end in order to respect the pageSize but to make it working right now you might consider the proposed fix